integrated_channels/canvas/client.py

```diff
@@ -44,7 +47,78 @@ def update_content_metadata(self, serialized_data):
     def delete_content_metadata(self, serialized_data):
         pass
 
-    def _get_oauth_access_token(self, client_id, client_secret):
+    def _post(self, url, data, scope):
+        """
+        Make a POST request using the session object to a Canvas endpoint.
+
+        Args:
+            url (str): The url to send a POST request to.
+            data (str): The json encoded payload to POST.
+        """
+        self._create_session()
+        response = self.session.post(url, data=data)
+        return response.status_code, response.text
+
+    def _delete(self, url, data, scope):
         """
-        TODO: get oauth token for canvas api access
+        Make a DELETE request using the session object to a Canvas endpoint.
+
+        Args:
+            url (str): The url to send a DELETE request to.
+            data (str): The json encoded payload to DELETE.
+        """
+        self._create_session()
+        response = self.session.delete(url, data=data)
+        return response.status_code, response.text
+
+    def _create_session(self):
         """
+        Instantiate a new session object for use in connecting with Canvas. Each enterprise customer
+        connecting to Canvas should have a single client session.
+        """
+        if self.session is None:
+            # Create a new session with a valid token
+            if self.session:
```

Comment:
wait how can this be? the if above says is None ... ? 😕

Reply:
:face_palm: good points